mazo.cpp

```diff
@@ -0,0 +1,123 @@
+#include"mazo.h"
+#include<iostream>
+/* ----- DEFINICIONES ----- */
+
+Mazo::Mazo(int cantidadDeCartasJuegaDoble, int cantidadDeCartasBloquearTurno, int cantidadDeCartasAgarrarCincoFichas, int cantidadDeCartasEliminarMazoJugadorSiguiente) {
+

```

Comment:
Como dije en algún comentario, acá falta la línea `this->mazo = new Cola<Carta*>();` para inicializar el atributo `this-mazo`, pero el caso es que igual este algoritmo, tal y como está, es muy costoso e ineficiente.
Por suerte Nico descubrió que lo que jodía tanto era el crear memoria dinámica para después borrarla si no se acolaba al mazo.
Acá te propongo una versión modificada que funciona perfecto:
```c++
Mazo::Mazo(int cantidadDeCartasJuegaDoble, int cantidadDeCartasBloquearTurno, int cantidadDeCartasAgarrarCincoFichas, int cantidadDeCartasEliminarMazoJugadorSiguiente) {

        mazo = new Cola<Carta*>();

	    int cantidadTotal = cantidadDeCartasJuegaDoble + cantidadDeCartasBloquearTurno + cantidadDeCartasAgarrarCincoFichas + cantidadDeCartasEliminarMazoJugadorSiguiente;

	    int contador1 = 0;
	    int contador2 = 0;
	    int contador3 = 0;
	    int contador4 = 0;

	    int cantidadDeCartas = 0;

	    while (cantidadDeCartas < cantidadTotal) {

	        CartaEspecial tipoAleatorio = generarNumeroRandom();
	        if (tipoAleatorio == JuegaDoble) {

	            if (contador1 < cantidadDeCartasJuegaDoble) {

	                mazo->acolar(new Carta(tipoAleatorio));
	                contador1++;
	                cantidadDeCartas++;

	            }
	        } else if (tipoAleatorio == BloquearTurno) {

	            if (contador2 < cantidadDeCartasBloquearTurno) {

	                mazo->acolar(new Carta(tipoAleatorio));
	                contador2++;
	                cantidadDeCartas++;

	            }
	        } else if (tipoAleatorio == AgarrarCincoFichas) {

	            if (contador3 < cantidadDeCartasAgarrarCincoFichas) {

	                mazo->acolar(new Carta(tipoAleatorio));
	                contador3++;
	                cantidadDeCartas++;

	            }

	        } else if (tipoAleatorio == EliminarMazoJugadorSiguiente) {

	            if (contador3 < cantidadDeCartasEliminarMazoJugadorSiguiente) {

	                mazo->acolar(new Carta(tipoAleatorio));
	                contador4++;
	                cantidadDeCartas++;

	            }
	        }
	    }

}
```

También como comentario extra, noté que tenés un método `acolarCarta()` en el TDA Mazo. Si lo usás queda más piola porque no se ve el atributo interno `this->mazo`:
```c++
Mazo::Mazo(int cantidadDeCartasJuegaDoble, int cantidadDeCartasBloquearTurno, int cantidadDeCartasAgarrarCincoFichas, int cantidadDeCartasEliminarMazoJugadorSiguiente) {

        mazo = new Cola<Carta*>();

	    int cantidadTotal = cantidadDeCartasJuegaDoble + cantidadDeCartasBloquearTurno + cantidadDeCartasAgarrarCincoFichas + cantidadDeCartasEliminarMazoJugadorSiguiente;

	    int contador1 = 0;
	    int contador2 = 0;
	    int contador3 = 0;
	    int contador4 = 0;

	    int cantidadDeCartas = 0;

	    while (cantidadDeCartas < cantidadTotal) {

	        CartaEspecial tipoAleatorio = generarNumeroRandom();
	        if (tipoAleatorio == JuegaDoble) {

	            if (contador1 < cantidadDeCartasJuegaDoble) {

	                this->acolarCarta(new Carta(tipoAleatorio));
	                contador1++;
	                cantidadDeCartas++;

	            }
	        } else if (tipoAleatorio == BloquearTurno) {

	            if (contador2 < cantidadDeCartasBloquearTurno) {

	                this->acolarCarta(new Carta(tipoAleatorio));
	                contador2++;
	                cantidadDeCartas++;

	            }
	        } else if (tipoAleatorio == AgarrarCincoFichas) {

	            if (contador3 < cantidadDeCartasAgarrarCincoFichas) {

	                this->acolarCarta(new Carta(tipoAleatorio));
	                contador3++;
	                cantidadDeCartas++;

	            }

	        } else if (tipoAleatorio == EliminarMazoJugadorSiguiente) {

	            if (contador3 < cantidadDeCartasEliminarMazoJugadorSiguiente) {

	                this->acolarCarta(new Carta(tipoAleatorio));
	                contador4++;
	                cantidadDeCartas++;

	            }
	        }
	    }

}
```